rpm/filescanner.go

```diff
@@ -0,0 +1,89 @@
+package rpm
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"runtime/trace"
+
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+)
+
+const (
+	scannerName    = "rpm"
+	scannerVersion = "1"
+	scannerKind    = "file"
+)
+
+var (
+	_ indexer.FileScanner      = (*FileScanner)(nil)
+	_ indexer.VersionedScanner = (*FileScanner)(nil)
+)
+
+// FileScanner implements [indexer.FileScanner], it examines RPM
+// databases and reports installed files.
+type FileScanner struct{}
+
+// Name implements [indexer.VersionedScanner].
+func (*FileScanner) Name() string { return scannerName }
+
+// Version implements [indexer.VersionedScanner].
+func (*FileScanner) Version() string { return scannerVersion }
+
+// Kind implements [indexer.VersionedScanner]
+func (*FileScanner) Kind() string { return scannerKind }
+
+// Scan reports any found Files that were installed via RPMs in the
+// layer.
+//
+// It's an expected outcome to return (nil, nil) no RPM packages are found in the Layer.
+func (s *FileScanner) Scan(ctx context.Context, layer *claircore.Layer) ([]claircore.File, error) {
+	if err := ctx.Err(); err != nil {
```

Comment:
just curious, how come this was added?

Reply:
To early out of an expensive function if the context is cancelled.